### PR TITLE
[DEV-10945] Update the disaster award endpoints to use the covid_faba_spending table

### PR DESCRIPTION
**Pull request by aguest-kc**
**Description:**
Updates the `/api/v2/disaster/award/amount/` and `/api/v2/disaster/award/count/` endpoints to use the `covid_faba_spending` table for their results.

**Technical details:**
A new view was added that groups Covid-19 FABA spending by award and then stores those totals and counts in the `covid_faba_spending` table. The disaster award endpoints then use these rows for their calculations.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
3. [x] Necessary PR reviewers:
    - [x] Backend
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-10945](https://federal-spending-transparency.atlassian.net/browse/DEV-10945):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download)
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
2. API documentation updated
No API documentation was affected by this change.

4. Matview impact assessment completed
No matviews are affected by this change.

5. Frontend impact assessment completed
The frontend is not impacted by this change.
```

### CODE REVIEW

File: usaspending_api/disaster/v2/views/award/amount.py [hunk 1]
```diff
@@ -40,13 +47,41 @@ def post(self, request):
...
+                total_obligation_sum=Sum("obligation_sum"),
+                total_outlay_sum=Sum("outlay_sum"),
+                total_face_value_of_loan=Sum("face_value_of_loan"),
+            )
+        )
+
+        if self.award_type_codes:
+            queryset = queryset.filter(award_type__in=self.award_type_codes)
+
+        if award_type and award_type.lower() == "procurement":
```
Comment (dporth-frb) [inline usaspending_api/disaster/v2/views/award/amount.py:66]: award_type is not a boolean. So coding `if award_type` to check if the variable equals `None` is implicit. This predicate can be improved to be more explicit. Try `if award_type is not None`

File: usaspending_api/disaster/v2/views/award/amount.py [hunk 2]
```diff
@@ -40,13 +47,41 @@ def post(self, request):
...
+            queryset = queryset.filter(award_type__in=procurement_type_mapping.keys())
+        elif award_type and award_type.lower() == "assistance":
```
Comment (dporth-frb) [inline usaspending_api/disaster/v2/views/award/amount.py:68]: Same comment here.

File: usaspending_api/disaster/v2/views/award/amount.py [hunk 3]
```diff
@@ -40,13 +47,41 @@ def post(self, request):
...
+            queryset = queryset.filter(award_type__in=assistance_type_mapping.keys())
+
+        result = {
+            "award_count": sum([row.total_award_count for row in queryset]),
+            "obligation": sum([row.total_obligation_sum for row in queryset]),
+            "outlay": sum([row.total_outlay_sum for row in queryset]),
+        }
+
+        # Add face_value_of_loan if any loan award types were included in the request
+        if self.award_type_codes and any(
```
Comment (dporth-frb) [inline usaspending_api/disaster/v2/views/award/amount.py:78]: Same comment here, except now the variable is `award_type_codes`

File: usaspending_api/disaster/v2/views/award/amount.py [hunk 4]
```diff
@@ -40,13 +47,41 @@ def post(self, request):
...
+            .filter(defc__in=self.filters["def_codes"])
+            .annotate(
+                total_award_count=Sum("award_count"),
+                total_obligation_sum=Sum("obligation_sum"),
+                total_outlay_sum=Sum("outlay_sum"),
+                total_face_value_of_loan=Sum("face_value_of_loan"),
+            )
+        )
+
+        if self.award_type_codes:
```
Comment (dporth-frb) [inline usaspending_api/disaster/v2/views/award/amount.py:63]: Same comment here, except now the variable is award_type_codes

**Review by dporth-frb — CHANGES_REQUESTED** [review]

File: usaspending_api/disaster/v2/views/award/amount.py (hunk 1 above)
Comment (aguest-kc) [reply usaspending_api/disaster/v2/views/award/amount.py]: Updated the `None` checks to be explicit.

**Review by dporth-frb — APPROVED** [review]

**Pull request merged**